src/app/pages/quickMaps/pages/locationSelect/locationSelect.component.ts

```diff
@@ -36,58 +50,143 @@ export class LocationSelectComponent implements OnInit, AfterViewInit {
-              },
-              mouseout: () => {
-                this.resetHighlight(layer);
-              },
-              click: (e) => {
-                this.map.fitBounds(e.target.getBounds());
-                window.alert(`you clicked on ${feature.properties.sovereignt}`);
-              },
-            });
-          },
-        }).addTo(this.map);
-      });
+    this.addCountriesMapLayer();
+
+    if (this.activeCountry) {
+      this.selectFeature(this.activeCountry);
+    }
+  }
+
+  private resetHighlight(layer: L.Layer): void {
+    if (layer !== this.selectedCountry) {
+      this.geojson.resetStyle(layer);
+    }
   }
 
   // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-  public highlightFeature(layer: any): void {
+  private hoverHighlightFeature(layer: any): void {
+    if (layer !== this.selectedCountry) {
+      layer.setStyle({
+        weight: 5,
+        color: '#9B51E0',
+        dashArray: '',
+        fillOpacity: 0.3,
+      });
+
+      if (!L.Browser.ie && !L.Browser.edge) {
+        layer.bringToFront();
+      }
+    }
+  }
+
+  private selectFeature(layer: any): void {
+    if (null != this.selectedCountry) {
+      this.selectedCountry.setStyle({
+        fillColor: '#8a66ad',
+        fillOpacity: 0.1,
+        color: '#1D3557',
+        opacity: 0.8,
+      });
+    }
+
+    this.selectedCountry = layer; // eslint-disable-line no-use-before-define
     layer.setStyle({
       weight: 5,
-      color: '#666',
+      color: '#703AA3',
       dashArray: '',
-      fillOpacity: 0.2,
+      fillOpacity: 0.3,
     });
 
     if (!L.Browser.ie && !L.Browser.edge) {
       layer.bringToFront();
     }
   }
 
-  public resetHighlight(layer: L.Layer): void {
-    this.geojson.resetStyle(layer);
+  private addCountriesMapLayer(): void {
+    this.dictionaryService.getDictionary(DictionaryType.COUNTRIES).then((dict: Dictionary): void => {
+      const featureCollection: GeoJSON.FeatureCollection = {
+        type: 'FeatureCollection',
+        features: dict
+          .getItems<CountryDictionaryItem>()
+          .map((item) => item.geoFeature)
+          .filter((item) => null != item),
+      };
+      this.geojson = L.geoJSON(featureCollection, {
+        style: () => { // eslint-disable-line no-use-before-define
+          return {
+            fillColor: '#8a66ad',
+            fillOpacity: 0.1,
+            color: '#1D3557',
+            opacity: 0.8,
+          };
+        },
+        onEachFeature: (feature, singleFeatureLayer: L.Layer) => {
+          singleFeatureLayer.on({
+            mouseover: () => {
+              this.hoverHighlightFeature(singleFeatureLayer);
+            },
+            mouseout: () => {
+              this.resetHighlight(singleFeatureLayer);
+            },
+            click: (e) => {
+              this.quickMapsService.setCountryId(`${feature.properties.countryId}`);
+              this.selectFeature(e.target);
```

Comment:
Do you need this?  It should select via the observable subscription.

Reply:
That's for highlighting the selected country. Is there a way of doing it without it?

Reply:
I think it should work without that line as the line above should trigger it.